docs/proposals/advanced-nginx-extensions.md

```diff
@@ -0,0 +1,28 @@
+# Enhancement Proposal-2035: Advanced NGINX Extensions
+
+- Issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/2035
+- Status: Provisional
+
+## Summary
+
+NGINX Gateway Fabric (NGF) [exposes](/site/content/overview/gateway-api-compatibility.md) NGINX features via Gateway API
+resources (like HTTPRoute) and [NGINX extensions](nginx-extensions.md) (like ClientSettingPolicy). Combined, they
+expose a subset of the most common NGINX configuration. As we implement more Gateway API resources and NGINX extensions,
+the subset will grow. However, it will take time. Additionally, because the number of NGINX configuration directives
+and parameters is huge, not all of them will be supported that way. As a result, users are not able to implement certain
+NGINX use cases. To allow them to implement those use cases, we need to bring a new extension mechanism to NGF.
+
+## Goals
+
+- Allow users to insert NGINX configuration not supported via Gateway API resources or NGINX extensions.
+- Allow users to customize supported NGINX configuration (for example, add a parameter to an NGINX directive).
+- Support configuration from modules not loaded in NGINX by default or third-party modules.
+- Most of the configuration complexity should fall onto the cluster operator persona, not the application developer.
```

Comment:
What is meant by this?

That we should only allow NGINX extensions at a Gateway level?
That application developers will not need to use advanced NGINX extensions?
That although application developers and cluster operators can both make use of advanced NGINX extensions, most complexity will happen to manifest when cluster operators implement?

Reply:
I was thinking there were two parts:
(1) Creating an advanced extension. This will require knowledge of NGF and NGINX - how to create an extension to generate correct NGINX configuration for a particular use case.
(2) Using an extension. Example - creating a Policy with a few values configured and target HTTPRoute with it.

So
(1) requires knowledge of how to create an extension and how to configure NGINX
(2) only requires knowledge on how to use particular extension (let's say extension Policy has 3 fields - A, B and C - so the user need to know how to use those fields)

So (1) is complex and should fall onto cluster operator while (2) should be very easy, so app developers can easily use extensions.